### ISSUE TITLE
Seek with F7/F9

### ISSUE THREAD

**Issue by khesed** [iina/iina#4631]
**What you want IINA to do:**
I would like IINA to seek with F7/F9 media keys instead of control the playlist.

**What IINA does currently:**
F7/F9 media keys control the playlist.

**Why you think this should be added:**
There are no shortcuts to control seeking by one button press without focusing IINA or hovering the player and scrolling.

Personally, I haven't ever found `playlist-next`/`prev` to be useful when viewing videos, but find myself seeking frequently. If I am viewing a video and typing notes, it is easier to press F7 to rewind and relisten to something I missed than to change window focus or move the mouse.

**Comment by low-batt** [comment]
IINA adheres to Apples [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/), so for any new feature we always must check to see if the HIG provides any relevant guidance.

From a quick search I did not find anything in the HIG about the `Now Playing` feature. The [Keyboards](https://developer.apple.com/design/human-interface-guidelines/keyboards#Keyboard-shortcuts) section provides this guidance:
> **In general, don’t repurpose standard keyboard shortcuts for custom actions.** People can get confused when the shortcuts they know work differently in your app. Only consider redefining a standard shortcut if its action doesn’t make sense in your app. For example, an app that doesn’t support text editing doesn’t need a text-styling command like Italic, so it might use Command–I for an action that has more relevance in the app, such as Get Info.

Not clear that guidance would apply in this case.

I'm leaving it to the senior developers to figure out how this should be implemented.

**Comment by khesed** [comment]
Thank you for that. I see now my initial post was ambiguous.

> Not clear that guidance would apply in this case.

I should clarify that I think it best if this were an opt-in change, where a user could repurpose a standard shortcut. I believe the default should remain as it is now.

**Comment by low-batt** [comment]
Yes. Something like the settings available on the `Control` tab where you can specify the behaviors of trackpad gestures and mouse scrolling and clicking.

As I'm a newbie junior IINA developer I defer to the senior developers on the best way to add such a feature.

**Comment by purplespider** [comment]
This should be possible to do via the app's "Key Bindings" Settings, but it doesn't appear to work:

1. Duplicate a Default set.
2. Locate the `NEXT` key, and change the value to `seek 60`
3. Locate the `PREV` key, and change the value to `seek -60`

![Screenshot 2023-11-13 at 12 30 47@2x](https://github.com/iina/iina/assets/329880/240fb3a3-708d-499e-8ca3-fe7f2c25a3f6)

This should seemingly make the keyboard media keys now seek forwards and backwards 60 secs. But it doesn't seem to work for me.

**Comment by rahbari** [comment]
This is really useful especially when watching video in PIP mode.  For now you must first exit PIP then seek. 
Maybe Next and Prev buttons can become forward and rewind when watching video.

With Keychron keyboard there are two set of media controls:
<img width="830" alt="image" src="https://github.com/iina/iina/assets/163890/f78e7f69-0ce0-4195-8dae-b487fbd13ab4">
KC_MPRV & KC_MRWD both goes to previous track
KC_MNXT & KC_MFFD both goes to next track
maybe KC_MFFD & KC_MRWD should seek instead of track change.

**Comment by KizzyCode** [comment]
> IINA adheres to Apples [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/), so for any new feature we always must check to see if the HIG provides any relevant guidance.
> 
> From a quick search I did not find anything in the HIG about the `Now Playing` feature. The [Keyboards](https://developer.apple.com/design/human-interface-guidelines/keyboards#Keyboard-shortcuts) section provides this guidance:
> 
> > **In general, don’t repurpose standard keyboard shortcuts for custom actions.** People can get confused when the shortcuts they know work differently in your app. Only consider redefining a standard shortcut if its action doesn’t make sense in your app. For example, an app that doesn’t support text editing doesn’t need a text-styling command like Italic, so it might use Command–I for an action that has more relevance in the app, such as Get Info.
> 
> Not clear that guidance would apply in this case.

Just as note: Apple itself regularly repurposes those buttons for 15s-jumps, e.g. in Podcasts or Safari or Books (even by default). So unless explicitly stated otherwise, I think it's safe to assume that this would be 100% compliant with HIG 😊 
(Depending on your daily use, IINA might even be considered the outlier here; my brain is so hardwired to "everything that is not music just seeks" and then I watch something in IINA and well 😅)

**Comment by Jugibur** [comment]
+1 for having this feature working with customized "Key Bindings" settings!

**Comment by benniekiss** [comment]
Is it possible to implement this using the new plugin system?
